src/Shipment/Response/GetShipmentsResponse.php

```diff
@@ -50,14 +51,19 @@ private function decodeShipment(array $data): Shipment
 
         $options            = $data['options'] ?? [];
         $physicalProperties = $data['physical_properties'] ?? [];
+        $carrier            = new Carrier(['id' => $data['carrier_id']]);
 
         return new Shipment([
             'id'                       => $data['id'],
             'shopId'                   => $data['shop_id'],
             'barcode'                  => $data['barcode'],
             'carrier'                  => [
-                'subscriptionId' => $data['contract_id'],
-                'id'             => $data['carrier_id'],
+                'carrier' => [
+                    'subscriptionId'     => $data['contract_id'],
+                    'id'                 => $data['carrier_id'],
+                    'name'               => $carrier->name,
+                    'externalIdentifier' => $carrier->name,
```

Comment:
dit moet niet name zijn, en hoef je niet in te vullen, gaat vanzelf op basis van carrier name en contract id